DataSets/DataSets/Program.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Data;
+
+namespace DataSets
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            using (var db = new DB("Data Source=.\\;Initial Catalog=DataSetsDB;Integrated Security=True"))
+            {
+                Console.WriteLine("SELECT *:");
+                printEmailValues(db);
+                Console.WriteLine();
+
+                Console.WriteLine("After INSERT:");
+                db.InsertIntoEmailTable();
+                printEmailValues(db);
+                Console.WriteLine();
+
+                Console.WriteLine("After UPDATE:");
+                db.UpdateEmailTable();
+                printEmailValues(db);
+                Console.WriteLine();
+
+                Console.WriteLine("After DELETE:");
+                db.DeleteFromEmailTable();
+                printEmailValues(db);
+                Console.WriteLine();
+            }
+        }
+
+        private static void printEmailValues(DB db)
+        {
+            var sqlReader = db.ReadEmailTable();
```

Comment:
more logical will be to get objects instead of sql object.
since you created db class, we have to give for him then all responsibilities to deal with sql requests. Now i see here that DB class did part of the job just.

You can also create some own custom reader(EmailReader or EmailRepository) in which you can pass the DB instance, then custom reader will read and return some object in which will be all data

Reply:
My point that Program.cs should don't have responsibility to know in which index record has what

Reply:
I'm totally agree with you, fixed this in ac99eb15389d191f66c236016287479137beedbf